src/commands/generate/types.ts

```diff
@@ -69,24 +68,55 @@ export default class GenerateTypes extends Base {
         'include-descriptions': Flags.boolean({
             description:
                 'Include variable descriptions in the variable information comment',
+            default: true,
+        }),
+        obfuscate: Flags.boolean({
+            description: 'Obfuscate the variable keys.',
             default: false,
         }),
     }
     authRequired = true
+    methodNames: Record<string, string[]> = {}
+    orgMembers: OrganizationMember[]
+    includeDescriptions = false
+    inlineComments = false
+    obfuscate = false
+    project: Project
 
     public async run(): Promise<void> {
         const { flags } = await this.parse(GenerateTypes)
-        const { project, headless } = flags
-        await this.requireProject(project, headless)
+        const {
+            project,
+            headless,
+            'include-descriptions': includeDescriptions,
+            'inline-comments': inlineComments,
+            obfuscate,
+        } = flags
+        this.includeDescriptions = includeDescriptions
+        this.inlineComments = inlineComments
+        this.obfuscate = obfuscate
+        this.project = await this.requireProject(project, headless)
+
+        if (
+            this.project.settings.obfuscation.enabled &&
+            this.project.settings.obfuscation.required
+        ) {
+            this.obfuscate = true
```

Comment:
if `obfuscate` was explicitly passed as `false` it might be nice to log a warning that it's required and will be set to `true` regardless of the flag value